File: Quizup-Angular-Application/src/app/features/quiz/models/quizSubmission.ts
import { Answer } from './answer';

export class QuizSubmission {
  constructor(
    public id: string = '',
    public quizId: string = '',
    public studentId: string = '',
    public answers: Answer[],
    public submissionDate: Date = new Date(),
    public savedDate: Date = new Date(),
    public score: number = 0,
    public submissionStatus: SubmissionStatus = SubmissionStatus.IN_PROGRESS
  ) {}
}

enum SubmissionStatus {
  IN_PROGRESS = 'InProgress',
  SAVED = 'saved',
  SUBMITTED = 'submitted',
}

File: Quizup-Dotnet-Backend/QuizupAPI/Models/QuizSubmission.cs
namespace QuizupAPI.Models
{
    public class QuizSubmission
    {
        public long Id { get; set; }
        public long QuizId { get; set; }
        public Quiz? Quiz { get; set; }
        public long StudentId { get; set; }
        public Student? Student { get; set; }
        public DateTime SubmissionDate { get; set; }
        public DateTime? SavedDate { get; set; }
        public List<Answer>? Answers { get; set; }
        public string SubmissionStatus { get; set; } = "InProgress";
        public int? Score { get; set; }
    }
}